src/controllers/employer.ts

```diff
@@ -46,9 +48,7 @@ export const registerEmployer = async (
     })
 
     if (exists) {
-      return next(
-        new BadRequestError(`Email ${credential.email} already exists`)
-      )
+      next(new BadRequestError(`Email ${credential.email} already exists`))
```

Comment:
it should have **return** here, otherwise it will continue to execute the next blocks of code